### PR TITLE
chore: update developer documentation

### PR DESCRIPTION
**Pull request by simonpasquier**
This commit reorganizes the `docs/developer.md` page by separating the content into 3 main sections:
* Development setup
* Contribution guidelines
* Release management

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+# Developer Documentation
 
-## Dependencies
+## Development Tools
 
 The build system assumes the following binaries are in the `PATH`:
 
 ```
-make git go npm kind podman (or docker)
+git
```
Comment (jan--f) [inline docs/developer.md:12]: Should we mention `make` here?

Comment (simonpasquier) [reply docs/developer.md:12]: :facepalm: of course!

**Review by jan--f — APPROVED** [review]
Thanks a lot!
/lgtm

**Review by jan--f — APPROVED** [review]
/lgtm

**Pull request merged**